vm/vm.go

```diff
@@ -188,37 +196,29 @@ func (v *vm) Execute(txns []core.Transaction, declaredClasses []core.Class, bloc
 	paidFeesOnL1CStr := cstring(paidFeesOnL1Bytes)
 	txnsJSONCstr := cstring(txnsJSON)
 	classesJSONCStr := cstring(classesJSON)
+	contractInfoJSONCStr := cstring(contractInfoJSON)
 
 	sequencerAddressBytes := sequencerAddress.Bytes()
-	gasPriceWEIBytes := gasPriceWEI.Bytes()
 
-	if gasPriceSTRK == nil {
-		gasPriceSTRK = &felt.Zero
-	}
-	gasPriceSTRKBytes := gasPriceSTRK.Bytes()
+	gasPriceWEIBytes := getBytesOrSetDefault(gasPriceWEI)
+	gasPriceSTRKBytes := getBytesOrSetDefault(gasPriceSTRK)
+	daGasPriceWEIBytes := getBytesOrSetDefault(daGasPriceWEI)
+	daGasPriceFRIBytes := getBytesOrSetDefault(daGasPriceFRI)
 
-	var skipChargeFeeByte byte
-	if skipChargeFee {
-		skipChargeFeeByte = 1
+	if gasPriceWEI.IsZero() || gasPriceSTRK.IsZero() || daGasPriceWEI.IsZero() || daGasPriceFRI.IsZero() {
+		return nil, nil, errors.New("gas prices must be non-zero")
 	}
 
-	var skipValidateByte byte
-	if skipValidate {
-		skipValidateByte = 1
-	}
-
-	var errOnRevertByte byte
-	if errOnRevert {
-		errOnRevertByte = 1
-	}
-	var legacyTraceJSONByte byte
-	if legacyTraceJSON {
-		legacyTraceJSONByte = 1
-	}
+	skipChargeFeeByte := boolToByte(skipChargeFee)
+	skipValidateByte := boolToByte(skipValidate)
+	errOnRevertByte := boolToByte(errOnRevert)
+	legacyTraceJSONByte := boolToByte(legacyTraceJSON)
+	useKzgDAByte := boolToByte(useKzgDA)
```

Comment:
we don't need these to be separate variables anymore, we can inline them in the call to `cairoVMExecute` below.